src/bower_components/emby-webcomponents/imageuploader/imageuploader.js

```diff
@@ -1,177 +1,142 @@
-                break; // noop
+                break;
             default:
-                require(['toast'], function (toast) {
-                    toast(globalize.translate('sharedcomponents#MessageFileReadError'));
+                require(["toast"], function(toast) {
+                    toast(globalize.translate("sharedcomponents#MessageFileReadError"))
                 });
-                break;
         }
     }
 
     function setFiles(page, files) {
-
         var file = files[0];
-
-        if (!file || !file.type.match('image.*')) {
-            page.querySelector('#imageOutput').innerHTML = '';
-            page.querySelector('#fldUpload').classList.add('hide');
-            currentFile = null;
-            return;
+        if (!file || !file.type.match("image.*")) {
+            page.querySelector("#imageOutput").innerHTML = "";
+            page.querySelector("#fldUpload").classList.add("hide");
+            return void(currentFile = null);
         }
-
         currentFile = file;
-
-        var reader = new FileReader();
-
+        var reader = new FileReader;
         reader.onerror = onFileReaderError;
-        reader.onloadstart = function () {
-            page.querySelector('#fldUpload').classList.add('hide');
+        reader.onloadstart = function() {
+            page.querySelector("#fldUpload").classList.add("hide");
         };
-        reader.onabort = function () {
+        reader.onabort = function() {
             loading.hide();
-            console.log('File read cancelled');
+            console.log("File read cancelled");
         };
-
-        // Closure to capture the file information.
-        reader.onload = (function (theFile) {
-            return function (e) {
-
-                // Render thumbnail.
-                var html = ['<img style="max-width:100%;max-height:100%;" src="', e.target.result, '" title="', escape(theFile.name), '"/>'].join('');
-
-                page.querySelector('#imageOutput').innerHTML = html;
-                page.querySelector('#fldUpload').classList.remove('hide');
-            };
-        })(file);
-
-        // Read in the image file as a data URL.
+        reader.onload = function(theFile) {
+            return function(e) {
+                var html = '<img style="max-width:100%;max-height:100%;" src="';
+                html += e.target.result;
+                html += '" title="'
+                html += escape(theFile.name);
+                html += '"/>';
+                page.querySelector("#imageOutput").innerHTML = html;
+                page.querySelector("#fldUpload").classList.remove("hide");
+            }
+        }
+        (file);
         reader.readAsDataURL(file);
     }
 
     function onSubmit(e) {
-
+        var dlg = dom.parentWithClass(this, "dialog");
         var file = currentFile;
-
-        if (!file) {
+        if (!file || ("image/png" !== file.type && "image/jpeg" !== file.type && "image/jpeg" !== file.type)) {
```

Comment:
#1 isn't exactly a cleaned up version, it only changed the minified files to the proper upstream source, so any weirdness introduced in that PR is old Emby code as well :grinning:

I only commented on it because it stated you'd made changes to it. Did you base your work on an older commit perhaps?